### ISSUE TITLE
Rule Change: ignore type annotations in no-restricted-globals

### ISSUE THREAD

**Issue by sethamus** [eslint/eslint#19769]
### What rule do you want to change?

no-restricted-globals

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```ts
/* eslint no-restricted-globals: ["error", "Promise"] */

// Correct error on Promise.resolve
// Incorrect error on Promise<any>
const x: Promise<any> = Promise.resolve();

// Correct no error on window.Promise.resolve
// Incorrect error on Promise<any>
const x: Promise<any> = window.Promise.resolve();
```

### What does the rule currently do for this code?

The rule reports an error on the type annotation (`Promise<any>`)

### What will the rule do after it's changed?

With a new option (e.g. `ignoreTypeAnnotation: true`), the rule will ignore restricted globals used in type annotations

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by bradzacher** [comment]
It's worth noting that we have [`no-restricted-types`](https://typescript-eslint.io/rules/no-restricted-types) which fills this hole.

It doesn't limit to global types (eg if you ban `Promise` it'll also ban locally defined types named `Promise`). But generally speaking it covers the same cases.

**Comment by nzakas** [comment]
Sounds good, marking as accepted.
